src/js/pages/templates/Page.jsx

```diff
@@ -1,32 +1,16 @@
-import React, { useState, useEffect } from 'react';
+import React from 'react';
 
 import Navbar from 'js/components/layout/Navbar';
 import Breadcrumbs from 'js/components/layout/Breadcrumbs';
 import Footer from 'js/components/layout/Footer';
-
-import { getUser, fetchApi } from 'js/services/api';
-import { useAuth0 } from 'js/services/react-auth0-spa';
+import { useUser } from 'js/pages/UserContext';
 
 export default ({ breadcrumbs, children }) => {
-  const { loading, isAuthenticated, getTokenSilently } = useAuth0();
-  const [userState, setUser] = useState(null);
-
-  useEffect(() => {
-    if (loading || !isAuthenticated) {
-      return;
-    };
-
-    getTokenSilently()
-      .then(token => fetchApi(token, getUser))
-      .then(setUser)
-      .catch(e => setUser({})); // TODO (dpordomingo):
-    // if could not get the user, but Auth0 is in logged in state, it will set an empty user
-    // We'll handle errors later, but this is blocker to ensure that at least the user can logout.
-  }, [loading, isAuthenticated, getTokenSilently]);;
+  const userContext = useUser();
```

Comment:
Maybe just `user` or `userState`?

```suggestion
  const user = useUser();
```

Reply:
I initially named it as `user`, but it could be understood as a regular variable, not being used for re-rendering the component when it changes.
`userState` could help to understand that it is not a regular `user` variable, so it will trigger a re-render, but it confuses, in my opinion, the component `state` and the React `context`. That's because I preferred to suffix it with `Context` instead of the regular `State`, used for component state variables.

Reply:
> userState could help to understand that it is not a regular user variable, so it will trigger a re-render, but it confuses, in my opinion, the component state and the React context

I agree on this.

> That's because I preferred to suffix it with Context instead of the regular State, used for component state variables.

By thinking more about this it totally make sense. I don't know the best practice, but I'm wondering whether it makes even more sense to make the value of the context not the user itself, but an object with the user. So that we'll actually end up doing:

```
const { user } = userUser();
```

Reply:
As far as reading the variable you know that it comes from the outside, and it will be updated when it changes (because of a re-render), it lgtm both:
```jsx
const pageSize = 2;
const userContext = useUserContext(); 
// const {userContext} = useUserContext(); // whatever you prefer, idm.
const {prsState, setPRsState} = useState([]);
// ...
// many lines below...
<span>{userContext.name}</span>
{prsState.splice(0,pageSize).map(pr=>pr.title)}

```